client/components/header-cake/back.jsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import ObserveWindowSizeMixin from 'lib/mixins/observe-window-resize';
+import Gridicon from 'components/gridicon';
+
+export default React.createClass( {
+
+	displayName: 'HeaderCakeBack',
+
+	mixins: [ ObserveWindowSizeMixin ],
+
+	propTypes: {
+		onClick: PropTypes.func,
+		href: PropTypes.string,
+		text: PropTypes.string
+	},
+
+	getDefaultProps() {
+		return {
+			text: 'Back'
+		}
+	},
+
+	render() {
+		const { text, href, onClick } = this.props;
+		const hideText = window.innerWidth <= 480 && text.length >= 8;
```

Comment:
I would suggest moving `480` and `8` values into named constants for readability.